shared/src/platform/context.ts

```diff
@@ -18,6 +18,11 @@ export interface EndpointPair {
     expose: Endpoint
 }
 
+export interface ClosablePair {
+    pair: EndpointPair
```

Comment:
I'd rename this to `endpoints`, because it's more important _what_ this is a pair _of_ (semantic), as opposed to that it is a pair of _something_ (type). You actually renamed this in one of the destructurings to `endpoints` so that would avoid the rename there.
```suggestion
    endpoints: EndpointPair
```